File: project/frontend-services/libs/identity-management/mapper/src/lib/mappers/TenantMapper.ts
import {TenantDTO} from "@frontend-services/identity-management/dto";
import {TenantInfo} from "@frontend-services/identity-management/model";

export class TenantMapper {

  static dtoToModel(dto: TenantDTO): TenantInfo {
    return new TenantInfo(dto.oid, dto.name, dto.email);
  }

  static modelToDto(model: TenantInfo): TenantDTO {
    return {oid: model.id, name: model.name, email: model.email};
  }
}

File: project/backend-services/identity-management-backend/application/src/main/java/pt/sensae/services/identity/management/backend/application/mapper/tenant/TenantMapper.java
package pt.sensae.services.identity.management.backend.application.mapper.tenant;

import pt.sensae.services.identity.management.backend.application.model.tenant.*;
import pt.sensae.services.identity.management.backend.domainservices.model.tenant.*;

public interface TenantMapper {

    IdentityQuery dtoToCommand(IdentityTokenDTO dto, boolean isMicrosoft);

    AccessTokenDTO commandToDto(TenantResult result);

    IdentityCommand dtoToCommand(AccessTokenDTO dto);

    PlaceTenantInDomainCommand dtoToCommand(PlaceTenantInDomainDTO dto);

    RemoveTenantFromDomainCommand dtoToCommand(ExpelTenantFromDomainDTO dto);
    UpdateTenantProfileCommand dtoToCommand(UpdateTenantDTO dto);
    
    TenantDTO resultToDto(TenantResult result);

}
